datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -1434,7 +1516,7 @@ mod tests {
                 ],
                 else_expr: Some(Box::new(lit(true))),
             })),
-            col("c1").or(col("c1").or(col("c2")).not())
+            col("c1").or(col("c1").not().and(col("c2").not()))
```

Comment:
Can you please update the comments here as well?

Reply:
Yes, I'll update the comments! Thanks!